FaceTracking/ViewController.swift

```diff
@@ -52,16 +52,18 @@ class ViewController: UIViewController {
     }()
     
     lazy var previewLayer: AVCaptureVideoPreviewLayer? = {
+        
         var previewLay = AVCaptureVideoPreviewLayer(session: self.session!)
-        previewLay?.videoGravity = AVLayerVideoGravityResizeAspectFill
+        previewLay.videoGravity = AVLayerVideoGravity.resizeAspectFill
         
         return previewLay
     }()
     
     lazy var frontCamera: AVCaptureDevice? = {
-        guard let devices = AVCaptureDevice.devices(withMediaType: AVMediaTypeVideo) as? [AVCaptureDevice] else { return nil }
         
-        return devices.filter { $0.position == .front }.first
+        let discoverySession = AVCaptureDevice.DiscoverySession.init(deviceTypes: [AVCaptureDevice.DeviceType.builtInWideAngleCamera], mediaType: .video, position: AVCaptureDevice.Position.front)
+        
+        return discoverySession.devices.count > 0 ? discoverySession.devices.first : nil
```

Comment:
you can just return discoverySession.devices.first, its optional

Reply:
check